### PR TITLE
GameState Editor Fixes and Combat Debugger HP Expansion

### PR DESCRIPTION
**Pull request by Zychronix**

### CODE REVIEW

File: src/main/java/legend/game/debugger/CombatDebuggerController.java [hunk 1]
```diff
@@ -104,8 +104,8 @@ public ListItem fromString(final String string) {
 
     this.level.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(1, 9999));
     this.dlevel.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 5));
-    this.hp.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 65535));
-    this.maxHp.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(1, 65535));
+    this.hp.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(0, 2147483647));
+    this.maxHp.setValueFactory(new SpinnerValueFactory.IntegerSpinnerValueFactory(1, 2147483647));
```
Comment (Ink230) [inline src/main/java/legend/game/debugger/CombatDebuggerController.java:108]: Integer.MAX_VALUE or pester Monoxide for a master enum file

File: src/main/resources/legend/game/debugger/combat_debugger.fxml [hunk 2]
```diff
@@ -9,7 +9,7 @@
 <?import javafx.scene.layout.HBox?>
 <?import javafx.scene.layout.VBox?>
 
-<VBox maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="400.0" prefWidth="600.0" stylesheets="@script_debugger.css" xmlns="http://javafx.com/javafx/18" xmlns:fx="http://javafx.com/fxml/1" fx:controller="legend.game.debugger.CombatDebuggerController">
+<VBox maxHeight="-Infinity" maxWidth="-Infinity" minHeight="-Infinity" minWidth="-Infinity" prefHeight="400.0" prefWidth="600.0" stylesheets="@script_debugger.css" xmlns="http://javafx.com/javafx/17.0.2-ea" xmlns:fx="http://javafx.com/fxml/1" fx:controller="legend.game.debugger.CombatDebuggerController">
```
Comment (Ink230) [inline src/main/resources/legend/game/debugger/combat_debugger.fxml:12]: Downgrade? Confident in no issues with it?

Comment (Zychronix) [reply src/main/resources/legend/game/debugger/combat_debugger.fxml:12]: It should be version 17, I am not sure why the editor keeps changing it for me.

File: src/main/java/legend/game/debugger/CombatDebuggerController.java (hunk 1 above)
Comment (Zychronix) [reply src/main/java/legend/game/debugger/CombatDebuggerController.java]: Changed it to Integer.MAX_VALUE

**Review by Ink230 — DISMISSED** [review]
LGTM!

File: src/main/java/legend/game/debugger/GameStateEditorController.java [hunk 3]
```diff
@@ -378,40 +378,56 @@ public void setGoods() {
...
   public void setChestFlags() {
-    gameState_800babc8._1a4[this.getChestFlags.getSelectionModel().getSelectedIndex()] = Integer.valueOf(this.textChestFlags.getText());
+    gameState_800babc8.chestFlags_1c4[this.getChestFlags.getSelectionModel().getSelectedIndex()] = Integer.valueOf(this.textChestFlags.getText());
   }
 
   public void getEquipment() {
-    this.textEquipment.setText(String.valueOf(gameState_800babc8.equipment_1e8.getInt((int)this.getEquipment.getValue())));
+    if(gameState_800babc8.equipment_1e8.size() > 0) {
+      this.textEquipment.setText(String.valueOf(gameState_800babc8.equipment_1e8.getInt((int)this.getEquipment.getValue())));
+    }else{
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/GameStateEditorController.java:402]:
```suggestion
    } else {
```

File: src/main/java/legend/game/debugger/GameStateEditorController.java [hunk 4]
```diff
@@ -378,40 +378,56 @@ public void setGoods() {
...
+      this.textEquipment.setText(String.valueOf(255));
+    }
   }
 
   @FXML
   public void setEquipment() {
-    gameState_800babc8.equipment_1e8.set((int)this.getEquipment.getValue(), Integer.valueOf(this.textEquipment.getText()));
+    if((int)this.getEquipment.getValue() >= gameState_800babc8.equipment_1e8.size()) {
+      gameState_800babc8.equipment_1e8.add(Integer.valueOf(this.textEquipment.getText()));
+    }else{
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/GameStateEditorController.java:411]:
```suggestion
    } else {
```

File: src/main/java/legend/game/debugger/GameStateEditorController.java [hunk 5]
```diff
@@ -378,40 +378,56 @@ public void setGoods() {
...
+      gameState_800babc8.equipment_1e8.set((int)this.getEquipment.getValue(), Integer.valueOf(this.textEquipment.getText()));
+    }
   }
 
   public void getItems() {
-    this.textItems.setText(String.valueOf(gameState_800babc8.items_2e9.getInt((int)this.getItems.getValue())));
+    if(gameState_800babc8.items_2e9.size() > 0) {
+      this.textItems.setText(String.valueOf(gameState_800babc8.items_2e9.getInt((int)this.getItems.getValue())));
+    }else{
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/GameStateEditorController.java:419]:
```suggestion
    } else {
```

File: src/main/java/legend/game/debugger/GameStateEditorController.java [hunk 6]
```diff
@@ -378,40 +378,56 @@ public void setGoods() {
...
+      this.textEquipment.setText(String.valueOf(255));
+    }
   }
 
   @FXML
   public void setItems() {
-    gameState_800babc8.items_2e9.set((int)this.getItems.getValue(), Integer.valueOf(this.textItems.getText()));
+    if((int)this.getItems.getValue() >= gameState_800babc8.items_2e9.size()) {
+      gameState_800babc8.items_2e9.add(Integer.valueOf(this.textItems.getText()));
+    }else{
```
Comment (LordMonoxide) [inline src/main/java/legend/game/debugger/GameStateEditorController.java:428]:
```suggestion
    } else {
```

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**